File: app/src/main/java/cat/xojan/random1/Application.java
package cat.xojan.random1;

import com.crashlytics.android.Crashlytics;
import com.crashlytics.android.answers.Answers;
import com.squareup.leakcanary.LeakCanary;

import cat.xojan.random1.injection.component.AppComponent;
import cat.xojan.random1.injection.component.DaggerAppComponent;
import cat.xojan.random1.injection.module.AppModule;
import io.fabric.sdk.android.Fabric;

public class Application extends android.app.Application {

    protected AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        initCrashlytics();
        initInjector();
        initLeakDetection();
    }

    public AppComponent getAppComponent() {
        return mAppComponent;
    }

    private void initInjector() {
        mAppComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    private void initLeakDetection() {
        if (BuildConfig.DEBUG) {
            LeakCanary.install(this);
        }
    }

    private void initCrashlytics() {
        if (!BuildConfig.DEBUG) {
            Fabric.with(this, new Answers(), new Crashlytics());
        }
    }
}

File: app/src/main/java/cat/xojan/random1/Application.kt
package cat.xojan.random1

import android.support.multidex.MultiDexApplication
import cat.xojan.random1.injection.component.AppComponent
import cat.xojan.random1.injection.component.DaggerAppComponent
import cat.xojan.random1.injection.module.AppModule
import com.squareup.leakcanary.LeakCanary

class Application : MultiDexApplication() {

     lateinit var appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()
        initInjector()
        initLeakDetection()
    }

    private fun initInjector() {
        appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .build()
    }

    private fun initLeakDetection() {
        if (BuildConfig.DEBUG) {
            LeakCanary.install(this)
        }
    }
}